doc/examples/streamline_registration.py

```diff
@@ -0,0 +1,322 @@
+                        "transformed_sagittal.png")
+regtools.overlay_slices(static, warped_moving, None, 1, "Static", "Moving",
+                        "transformed_coronal.png")
+regtools.overlay_slices(static, warped_moving, None, 2, "Static", "Moving",
+                        "transformed_axial.png")
+
+"""
+.. figure:: transformed_sagittal.png
+   :align: center
+.. figure:: transformed_coronal.png
+   :align: center
+.. figure:: transformed_axial.png
+   :align: center
+
+   Deformable registration result.
+"""
+
+"""
+We read the streamlines from file in voxel space:
+
+"""
+
+from dipy.io.streamline import load_trk
+
+streamlines, hdr = load_trk('lr-superiorfrontal.trk')
+
+
+"""
+We then apply the obtained deformation field to the streamlines. We first
+apply the non-rigid warping and then apply the computed affine transformation
+whose extents must be corrected to account for the different voxel grids of
+the moving and static images:
+
+"""
+
+from dipy.tracking.streamline import deform_streamlines, Streamlines
+
+from dipy.tracking.utils import move_streamlines
+
+warped_streamlines = \
+    deform_streamlines(streamlines,
+                       deform_field=mapping.get_forward_field(),
+                       stream_to_current_grid=moving_affine,
+                       current_grid_to_world=mapping.codomain_grid2world,
+                       stream_to_ref_grid=static_affine,
+                       ref_grid_to_world=mapping.domain_grid2world)
+
+
+adjusted_affine = moving_affine.copy()
+adjusted_affine[0][3] = -adjusted_affine[0][3] - static_affine[0][3]
+adjusted_affine[1][3] = -adjusted_affine[1][3] + static_affine[1][3]
+adjusted_affine[2][3] = -adjusted_affine[2][3] + static_affine[2][3]
+
+rotation, scale = np.linalg.qr(affine_map.affine)
+streams = move_streamlines(streamlines, rotation)
+scale[0:3, 0:3] = np.dot(scale[0:3, 0:3], np.diag(1. / hdr['voxel_sizes']))
+scale[0:3, 3] = abs(scale[0:3, 3])*1. / hdr['voxel_sizes'][0]
+
+affine_streamlines = Streamlines(move_streamlines(warped_streamlines, scale))
+
+
+"""
+We display the original streamlines and the registered streamlines:
+
+"""
+
+from dipy.viz import has_fury
+
+def show_template_bundles(bundles, show=True, fname=None):
+
+    renderer = window.Renderer()
+    template_img_data = img_t2_mni.get_data().astype('bool')
+    template_actor = actor.contour_from_roi(template_img_data,
+                                            color=(50, 50, 50), opacity=0.05)
+    renderer.add(template_actor)
+    lines_actor = actor.streamtube(bundles, window.colors.orange,
+                                   linewidth=0.3)
+    renderer.add(lines_actor)
+    if show:
+        window.show(renderer)
+    if fname is not None:
+        sleep(1)
+        window.record(renderer, n_frames=1, out_path=fname, size=(900, 900))
+
+
+if has_fury:
+
+    from dipy.viz import window, actor
+
+    from time import sleep
```

Comment:
Why do you need `sleep` ?

Reply:
@jhlegarreta ?

Reply:
:+1: I guess it is not really needed. I'll remove that. I'd dare to say that it may be worthwhile having a look at its use in other examples that may not need it.